lib/htmlReporter.js

```diff
@@ -43,15 +48,39 @@ class HTMLReporter {
         appFileName: this.opts.appFileName,
         dumpAssets: getAssetsContent,
         duplicates: JSON.stringify({
-          data: this.duplicates.map(duplicate => {
-            return duplicate.map(({ node, id }) => {
-              return {
-                id,
-                sourceCode: node.sourceCode,
-                loc: node.loc
-              };
-            });
-          })
+          data: this.duplicates
+            .map(duplicate => {
+              return duplicate.map(({ node, id, start, end }) => {
+                // Get the actual file
+                const fileContent = fs.readFileSync(id).toString();
```

Comment:
file content sources can be extracted from CPA

https://github.com/vigneshshanmugam/js-cpa/blob/master/lib/cpa.js#L20